lib/core/test/integration/Test/Integration/Framework/DSL.hs

```diff
@@ -894,13 +895,43 @@ listWalletsViaCLI
 listWalletsViaCLI ctx = cardanoWalletCLI @t
     ["wallet", "list", "--port", show (ctx ^. typed @Port) ]
 
-updateWalletViaCLI
+updateWalletNameViaCLI
     :: forall t r s. (CmdResult r, KnownCommand t, HasType Port s)
     => s
     -> [String]
     -> IO r
-updateWalletViaCLI ctx args = cardanoWalletCLI @t
-    (["wallet", "update", "--port", show (ctx ^. typed @Port)] ++ args)
+updateWalletNameViaCLI ctx args = cardanoWalletCLI @t
+    (["wallet", "update", "name", "--port", show (ctx ^. typed @Port)] ++ args)
+
+updateWalletPassphraseViaCLI
+    :: forall t s. (KnownCommand t, HasType Port s)
+    => s
+    -> String
+        -- ^ Wallet id
+    -> String
+        -- ^ Old passphrase
+    -> String
+        -- ^ New passphrase
+    -> String
+        -- ^ New passphrase (repeated for confirmation)
```

Comment:
Small suggestion: avoid stringly-typed APIs as much as possible. A cheap alternative is often a record with named fields. It gets very readable at the call-site when used in explicit form.

Reply:
I completely agree with this.

In this case, I was attempting to be consistent with the surrounding module and similar functions, most of which are stringly-typed (for example, `createWalletViaCLI`).

Perhaps we create a further PR to go over the functions in this module and make the types clearer?

Reply:
> Small suggestion: avoid stringly-typed APIs as much as possible. A cheap alternative is often a record with named fields. It gets very readable at the call-site when used in explicit form.


I completely agree with this.

In this case, I was attempting to be consistent with the surrounding module and similar functions, most of which are stringly-typed (for example, createWalletViaCLI).

Perhaps we create a further PR to go over the functions in this module and make the types clearer?